plugins/PostProcessingPlugin/scripts/AddCoolingProfile.py

```diff
@@ -273,46 +280,141 @@ def getSettingDataString(self):
+                    "description": "The mainboard circuit number of the Chamber or Auxiliary Fan.",
+                    "type": "int",
+                    "unit": "#    ",
+                    "default_value": 0,
+                    "minimum_value": 0,
+                    "enabled": "enable_bv_fan and bv_fan_speed_control_enable"
+                },
+                "bv_fan_speed":
+                {
+                    "label": "    Chamber/Aux Fan Speed %",
+                    "description": "The speed of the Chamber or Auxiliary Fan.  This will be converted to PWM Duty Cycle (0-255) or (RepRap 0-1 if that is enabled in Cura).  If your specified fan does not operate on variable speeds then set this to '100'.",
+                    "type": "int",
+                    "unit": "%    ",
+                    "default_value": 50,
+                    "maximum_value": 100,
+                    "minimum_value": 0,
+                    "enabled": "enable_bv_fan and bv_fan_speed_control_enable"
+                },
+                "bv_fan_start_layer":
+                {
+                    "label": "    Chamber/Aux Fan Start Layer",
+                    "description": "The layer to start the Chamber or Auxiliary Fan.  Use the Cura preview layer number and the fan will start at the beginning of the layer.",
+                    "type": "int",
+                    "unit": "Layer#    ",
+                    "default_value": 1,
+                    "minimum_value": 1,
+                    "enabled": "enable_bv_fan and bv_fan_speed_control_enable"
+                },
+                "bv_fan_end_layer":
+                {
+                    "label": "    Chamber/Aux Fan End Layer",
+                    "description": "The layer number for Chamber or Auxiliary Fan to turn off.  Use the Cura preview layer number or '-1' to indicate the end of the print.  The fan will run until the end of the layer",
+                    "type": "int",
+                    "unit": "Layer#    ",
+                    "default_value": -1,
+                    "minimum_value": -1,
+                    "enabled": "enable_bv_fan and bv_fan_speed_control_enable"
+                },
+                "enable_bv_fan":
+                {
+                    "label": "Hidden setting",
+                    "description": "This is enabled when machine_heated_bed is true, and in turn this enables 'bv_fan_speed_control_enable'.",
+                    "type": "bool",
+                    "default_value": false,
+                    "enabled": false
                 }
             }
         }"""
 
     def initialize(self) -> None:
         super().initialize()
-        scripts = Application.getInstance().getGlobalContainerStack().getMetaDataEntry("post_processing_scripts")
+        global_stack = Application.getInstance().getGlobalContainerStack()
+        extruder = global_stack.extruderList
+        extruder_count = global_stack.getProperty("machine_extruder_count", "value")
```

Comment:
Since you're using these in the execute as well, it would be better to define as `self.` in the initialize and then reuse there in execute.
```suggestion
        self.global_stack = Application.getInstance().getGlobalContainerStack()
        self.extruder_list = self.global_stack.extruderList
        self.extruder_count = self.global_stack.getProperty("machine_extruder_count", "value")
```

Reply:
Completed